addons/mrp/models/product.py

```diff
@@ -70,6 +79,15 @@ def _compute_used_in_bom_count(self):
         for product in self:
             product.used_in_bom_count = self.env['mrp.bom'].search_count([('bom_line_ids.product_id', '=', product.id)])
 
+    def write(self, values):
+        res = super().write(values)
+        if 'active' in values:
```

Comment:
``` python3
if 'active' in values:
    self.with_context(active_test=False).filtered(lambda p: p.active != values['active'].write({
        'active': values['active']
    })
return super().write(values)
```
@yhu-odoo What do you think of this? It would avoid to reactive the BoM if the user write active=True on a BoM already active

Reply:
@amoyaux That make sense. But i think it should like this 
```suggestion
        if 'active' in values:
            self.filtered(lambda p: p.active != values['active']).with_context(active_test=False).variant_bom_ids.write({
                'active': values['active']
            })
        return super().write(values)
```
I'll also apply the change to RR

Reply:
@yhu-odoo Yep